### PR TITLE
:sparkles: kirakira effect normalization

### PR DESCRIPTION
**Pull request by ChiaoteNi**
## Descriptions:
- This is the extended PR of the https://github.com/cardinalblue/GPUImage3/pull/9; the original one has closed automatically by rebasing the branch with dropping some of the comments.
- Please feel free to review the last commit directly [♻️ refactor the normalization and rename the maxLength to targetDimen…](https://github.com/cardinalblue/GPUImage3/commit/27125ee447d2c63b7d1c9b7723a3d40054ddc74c), all other ones have already been reviewed in the original PR.

- Implement kirakira effect normalization based on the changes in the branch [kirakira_normalization] (https://github.com/cardinalblue/CBVisualEffectBuiltins/compare/dev...task/kirakira_normalization) in CBVisualEffectBuiltIn
- Also, based on the changes in the branch [kirakira_normalization](https://github.com/cardinalblue/CBVisualEffectBuiltins/compare/dev...task/kirakira_normalization), I modified some default values for the parameters as well

# Screenshots

<table>
<tr>
<th>Before</th>
<th>After</th>
</tr>
<tr>
<td>

IMG_9700

</td>
<td>

IMG_9697

</td>
</tr>
</table>

### CODE REVIEW

File: framework/Source/Kirakira.swift [hunk 1]
```diff
@@ -79,11 +81,15 @@ public class Kirakira: OperationGroup {
             self.sparkleAmount = sparkleAmount
             self.frameRate = frameRate
             self.blur = blur
+            self.targetDimension =             targetDimension
```
Comment (yyjim) [inline framework/Source/Kirakira.swift]: nit: spacing

File: examples/iOS/SimpleImageFilter/SimpleImageFilter/Kirakira_Glamour.json [hunk 2]
```diff
@@ -74,5 +74,13 @@
     "frameRate" : {
         "type" : "int",
         "value" : 60
-    }
+    },
+    "maxLength" : {
```
Comment (yyjim) [inline examples/iOS/SimpleImageFilter/SimpleImageFilter/Kirakira_Glamour.json]: Should we update this to `targetDimension`?

File: examples/iOS/SimpleImageFilter/SimpleImageFilter/Kirakira_Barbie.json
Comment (Vivienne0206) [inline examples/iOS/SimpleImageFilter/SimpleImageFilter/Kirakira_Barbie.json:1]: Should also change "maxLength" to "targetDimension"

**Review by Vivienne0206 — COMMENTED** [review]
Should update the parameter's name in JSON

File: examples/iOS/SimpleImageFilter/SimpleImageFilter/Kirakira_Glamour.json (hunk 2 above)
Comment (Vivienne0206) [reply examples/iOS/SimpleImageFilter/SimpleImageFilter/Kirakira_Glamour.json]: I think we can just use the default value

File: examples/iOS/SimpleImageFilter/SimpleImageFilter/Kirakira_Barbie.json
Comment (ChiaoteNi) [reply examples/iOS/SimpleImageFilter/SimpleImageFilter/Kirakira_Barbie.json:1]: Thanks🙏🏻🙏🏻

File: framework/Source/Kirakira.swift (hunk 1 above)
Comment (ChiaoteNi) [reply framework/Source/Kirakira.swift]: 🙏🏻

**Comment by ChiaoteNi** [comment]
Hi @yyjim and @Vivienne0206,
I've corrected the issues in JSON files. Please help review the PR again when you're available. Thanks!

File: framework/Source/Operations/CBRescaleEffect.swift [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+            return
+        }
+        guard let commandBuffer = sharedMetalRenderingDevice.commandQueue.makeCommandBuffer() else {
+            return
+        }
+
+        defer {
+            textureInputSemaphore.signal()
+        }
+        let _ = textureInputSemaphore.wait(timeout:DispatchTime.distantFuture)
```
Comment (yyjim) [inline framework/Source/Operations/CBRescaleEffect.swift]: suggest to change the order
```suggestion
        let _ = textureInputSemaphore.wait(timeout: DispatchTime.distantFuture)
        defer {
            textureInputSemaphore.signal()
        }
```

**Review by yyjim — APPROVED** [review]

**Review by Vivienne0206 — APPROVED** [review]

**Pull request merged**